src/main/java/no/nav/pto/veilarbportefolje/domene/Bruker.java

```diff
@@ -148,14 +147,61 @@ private Bruker addAktivitetUtlopsdato(String type, Timestamp utlopsdato) {
         return this;
     }
 
+    public void kalkulerSisteEndring(Map<String, Endring> siste_endringer, List<String> kategorier){
+        if(siste_endringer == null ){
+            return;
+        }
+
+        for (String kategori : kategorier) {
```

Comment:
maybe this can be replaced with something like this: 

`Optional<Endring> nyesteSisteEndring = kategorier.stream().filter(k -> siste_endringer.containsKey(k)).map(k -> siste_endringer.get(k)).max(Comparator.comparing(se -> toLocalDateTimeOrNull(se.getTidspunkt())));
if (nyesteSisteEndring.isPresent()){
            ...
 }
`

Reply:
I think that would be nice 😄 but how do we do it with sisteEndringKategori?